src/api/registerAndDeposit.js

```diff
@@ -68,6 +68,10 @@ module.exports = async (dvf, depositData, starkPublicKey, nonce, signature, cont
         }
       }
     })
+
+    if (dvf.dvfStarkProvider && dvf.dvfStarkProvider.getWalletType() === 'LEDGER') {
+      await dvf.token.provideContractData(null, tx.tokenAddress, tx.quantum)
```

Comment:
Didn't we want to remove the need for contract data on ledger users or is that not possible?

Reply:
So in order for users to not have to enable "contract data", we need to call some ledger utility methods, AFAIK that's because ledger device doesn't have enough memory to parse all the arguments at once
So this is needed in order for that to happen